File: src/main/java/hello/IO/FileReaderClass.java
package hello.IO;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.Scanner;

public class FileReaderClass {

    String fileName;

    public FileReaderClass(String fileName) {
        this.fileName = fileName;
    }

    public String useScanner() {

        String content = null;

        try {

            StringBuilder result = new StringBuilder();

            File file = new File(this.fileName);
            Scanner reader = new Scanner(file);

            while (reader.hasNextLine()) {
                result.append(reader.nextLine());
                result.append(System.lineSeparator());
            }

            System.out.print(result);
            content = result.toString();

        } catch (FileNotFoundException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }

        return content;
    }

    public String useFileReader() {

        String content = null;

        try {

            StringBuilder builder = new StringBuilder();
            FileReader reader = new FileReader(this.fileName);

            int c;
            while ((c = reader.read()) != -1) {
                builder.append((char)c);
            }

            System.out.print(builder);
            content = builder.toString();

        } catch (IOException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();
        }

        return content;

    }

}

File: src/main/java/hello/IO/FileReaderClass.kt
package hello.IO

import java.io.File
import java.io.FileNotFoundException
import java.io.FileReader
import java.io.IOException
import java.util.*

class FileReaderClass(var fileName: String) {
    @Throws(FileNotFoundException::class)
    fun useScanner(): String {
        val result = StringBuilder()
        val file = File(fileName)
        val reader = Scanner(file)
        while (reader.hasNextLine()) {
            result.append(reader.nextLine())
            result.append(System.lineSeparator())
        }
        print(result)
        return result.toString()
    }

    @Throws(IOException::class)
    fun useFileReader(): String {
        val builder = StringBuilder()
        val reader = FileReader(fileName)
        var c: Int
        while (reader.read().also { c = it } != -1) {
            builder.append(c.toChar())
        }
        print(builder)
        return builder.toString()
    }

}